docs/testplan/Generic-hash-test-plan.md

```diff
@@ -0,0 +1,314 @@
+# Generic Hash Test Plan
+
+## Related documents
+
+| **Document Name** | **Link** |
+|-------------------|----------|
+| SONiC Generic Hash | [https://github.com/sonic-net/SONiC/doc/hash/hash-design.md]|
+
+
+## 1. Overview
+The hashing algorithm is used to make traffic-forwarding decisions for traffic exiting the switch.
+It makes hashing decisions based on values in various packet fields, as well as on the hash seed value.
+The packet fields used by the hashing algorithm varies by the configuration on the switch.
+
+For ECMP, the hashing algorithm determines how incoming traffic is forwarded to the next-hop device.
+For LAG, the hashing algorithm determines how traffic is placed onto the LAG member links to manage
+bandwidth by evenly load-balancing traffic across the outgoing links.
+
+GH is a feature which allows user to configure which hash fields suppose to be used by hashing algorithm.
+GH provides global switch hash configuration for ECMP and LAG.
+
+
+## 2. Requirements
+
+### 2.1 The Generic Hash feature supports the following functionality:
+1. Ethernet packet hashing configuration with inner/outer IP frames
+2. Global switch hash configuration for ECMP and LAG
+3. Warm/Fast reboot
+
+### 2.2 This feature will support the following commands:
+
+1. config: set switch hash global configuration
+2. show: display switch hash global configuration
```

Comment:
can also show the switch hash capabilities, maybe it should also be described here

Reply:
Done